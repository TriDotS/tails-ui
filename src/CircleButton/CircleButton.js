import React from 'react'
import styled from 'styled-components'

const CircleButton = styled.button`
  padding: 10px;
  cursor: pointer;
  text-decoration: none;
  border-radius: 50%;
  outline: none;
  color: ${props => props.fontColor || 'white'};
  background: ${props =>
    props.hollow ? 'transparent' : props.backgroundColor || 'palevioletred'};
  border: ${props =>
    props.hollow ? `2px solid ${props.borderColor}` : 'none'};
  box-shadow: ${props =>
    props.shadowed
      ? 'rgba(0, 0, 0, 0.16) 0px 3px 10px, rgba(0, 0, 0, 0.23) 0px 3px 10px'
      : null};
  width: ${props => (props.fullWidth ? '100%' : null)};
  
  &:active {
    box-shadow: 0;
    transform: translateY(2px);
    filter: brightness(85%);
  }
`

export default ({
  children,
  onClick,
  type,
  fullWidth,
  backgroundColor,
  fontColor,
  shadowed,
  hollow,
  borderColor,
}) =>
  <CircleButton
    type={type}
    fullWidth={fullWidth}
    backgroundColor={backgroundColor}
    fontColor={fontColor}
    onClick={onClick}
    shadowed={shadowed}
    hollow={hollow}
    borderColor={borderColor}
  >
    {children}
  </CircleButton>
