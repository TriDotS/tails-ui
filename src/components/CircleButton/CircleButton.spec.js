import CircleButton from './CircleButton'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<CircleButton />, div)
})

it('generate CircleButton snapshot', () => {
  const component = renderer.create(<CircleButton />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
