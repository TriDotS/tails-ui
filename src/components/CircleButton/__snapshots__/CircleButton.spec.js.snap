// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate CircleButton snapshot 1`] = `
<button
  className="sc-bdVaJa cWBayE"
  onClick={undefined}
  type={undefined}
/>
`;
